core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala

```diff
@@ -44,25 +44,24 @@ final case class InfoAppenders(info: ChunkSetInfo, appenders: TimeSeriesPartitio
  *     This allows for safe and cheap write buffer churn without losing any data.
  *   switchBuffers() is called before flush() is called in another thread, possibly.
  *
- * The main data structure used is the "infoMap" - an OffheapLFSortedIDMap, an extremely efficient, offheap sorted map
- * Note that other than the variables used in this class, there is NO heap memory used for managing chunks.  Thus
- * the amount of heap memory used for a partition is O(1) constant regardless of the number of chunks in a TSPartition.
- * The partition key and infoMap are both in offheap write buffers, and chunks and chunk metadata are kept in
- * offheap block memory.
+ * The main data structure used in inherited from ChunkMap, an efficient, offheap sorted map.
+ * Note that other than the variables used in this class, there is NO JVM-maneged memory used
+ * for managing chunks.  Thus the amount of JVM-managed memory used for a partition is constant
+ * regardless of the number of chunks in a TSPartition. The partition key and infoMap are both
+ * in offheap write buffers, and chunks and chunk metadata are kept in offheap block memory.
+ *
+ * Note: Inheritance is chosen over composition to avoid an extra object allocation, which
+ * speeds up GC and reduces memory overhead a bit.
  */
 class TimeSeriesPartition(val partID: Int,
                           val dataset: Dataset,
                           partitionKey: BinaryRegion.NativePointer,
                           val shard: Int,
                           bufferPool: WriteBufferPool,
                           val shardStats: TimeSeriesShardStats,
-                          // Volatile pointer to infoMap structure.  Name of field MUST match mapKlazz method above
-                          var mapPtr: BinaryRegion.NativePointer,
-                          // Shared class for mutating the infoMap / OffheapLFSortedIDMap given mapPtr above
-                          offheapInfoMap: OffheapLFSortedIDMapMutator,
-                          // Lock state used by OffheapLFSortedIDMap.
-                          var lockState: Int = 0)
-extends ReadablePartition with MapHolder {
+                          memFactory: MemFactory,
+                          initMapSize: Int)
```

Comment:
I wish there was a way not to have to pass in initMapSize.  It's kinda useless field after initialization.